conformance/utils/kubernetes/apply.go

```diff
@@ -139,6 +139,16 @@ func (a Applier) prepareGateway(t *testing.T, uObj *unstructured.Unstructured) {
 		err = unstructured.SetNestedSlice(uObj.Object, primOverlayAddrs, "spec", "addresses")
 		require.NoError(t, err, "could not overlay static addresses on Gateway %s/%s", ns, name)
 	}
+
+	//  This is being done in order to support the injection of implementation-specific address types
+	// into the test suite
+	if len(gwspec.Addresses) > 0 && *gwspec.Addresses[0].Type == "PLACEHOLDER_ADDRESS_TYPE" {
```

Comment:
Instead of hardcoding the PLACEHOLDER index to 0, can we look up for it? This would make this code future-proof and more robust.